src/swaps/bitcoin.rs

```diff
@@ -400,27 +400,36 @@ impl BtcSwapScript {
         Ok((script_balance.confirmed, script_balance.unconfirmed))
     }
 
-    /// Fetch (utxo,amount) pair for the script_pubkey of this swap.
-    /// Returns None if no utxo for the script_pubkey is found.
-    pub fn fetch_utxo(
+    /// Fetch (utxo,amount) pairs for all utxos of the script_pubkey of this swap.
+    pub fn fetch_utxos(
         &self,
         network_config: &ElectrumConfig,
-    ) -> Result<Option<(OutPoint, TxOut)>, Error> {
+    ) -> Result<Vec<(OutPoint, TxOut)>, Error> {
         let electrum_client = network_config.build_client()?;
         let spk = self.to_address(network_config.network())?.script_pubkey();
         let utxos = electrum_client.script_list_unspent(spk.as_script())?;
+        let utxo_pairs = utxos
+            .iter()
+            .map(|unspent| {
+                let outpoint = OutPoint::new(unspent.tx_hash, unspent.tx_pos as u32);
+                let txout = TxOut {
+                    script_pubkey: spk.clone(),
+                    value: Amount::from_sat(unspent.value),
+                };
+                (outpoint, txout)
+            })
+            .collect();
+        Ok(utxo_pairs)
+    }
 
-        if utxos.len() == 0 {
-            // No utxo found. Return None.
-            return Ok(None);
-        } else {
-            let outpoint_0 = OutPoint::new(utxos[0].tx_hash, utxos[0].tx_pos as u32);
-            let txout = TxOut {
-                script_pubkey: spk,
-                value: Amount::from_sat(utxos[0].value),
-            };
-            Ok(Some((outpoint_0, txout)))
-        }
+    /// Fetch the first (utxo,amount) pair for the script_pubkey of this swap, which is typically
+    /// the lockup utxo. Returns None if no utxo for the script_pubkey is found.
+    pub fn fetch_utxo(
```

Comment:
Should we get rid of this function completely?

Reply:
Done: 82c6bfb526cba84cb6e1273dcb143b078b5be1ea